packages/forms-web-app/__tests__/unit/controllers/examination/submitting-for/controller.test.js

```diff
@@ -0,0 +1,121 @@
+const {
+	getSubmittingFor,
+	postSubmittingFor
+} = require('../../../../../src/controllers/examination/submitting-for/controller');
+
+const {
+	getPageData
+} = require('../../../../../src/controllers/examination/submitting-for/utils/get-page-data');
+const {
+	getRedirectUrl
+} = require('../../../../../src/controllers/examination/submitting-for/utils/get-redirect-url');
+const {
+	setDeadlineDetailsSubmittingFor
+} = require('../../../../../src/controllers/examination/session/deadline');
+
+jest.mock('../../../../../src/controllers/examination/submitting-for/utils/get-page-data', () => ({
+	getPageData: jest.fn()
+}));
+jest.mock(
+	'../../../../../src/controllers/examination/submitting-for/utils/get-redirect-url',
+	() => ({
+		getRedirectUrl: jest.fn()
+	})
+);
+jest.mock('../../../../../src/controllers/examination/session/deadline', () => ({
+	setDeadlineDetailsSubmittingFor: jest.fn()
+}));
+
+describe('controllers/examination/submitting-for/controller', () => {
+	const req = {
+		body: {},
+		session: { examination: {} },
+		query: { mock: 'query' }
+	};
+	const res = {
+		redirect: jest.fn(),
+		render: jest.fn(),
+		status: jest.fn(() => res)
+	};
+	describe('#getSubmittingFor', () => {
+		describe('When rendering the submitting for page', () => {
+			describe('and there are no issues', () => {
+				beforeEach(() => {
+					getPageData.mockReturnValue({ mock: 'page data' });
+					getSubmittingFor(req, res);
+				});
+				it('should get the page data', () => {
+					expect(getPageData).toHaveBeenCalledWith(req.query, req.session);
+				});
+				it('should render the submitting for page with the page data', () => {
+					expect(res.render).toHaveBeenCalledWith('pages/examination/submitting-for', {
+						mock: 'page data'
+					});
+				});
+			});
+			describe('and there is an issue', () => {
+				beforeEach(() => {
+					getPageData.mockImplementation(() => {
+						throw new Error('something went wrong');
+					});
+					getSubmittingFor(req, res);
+				});
+				it('should render the Error page', () => {
+					expect(res.status).toHaveBeenCalledWith(500);
+					expect(res.render).toHaveBeenCalledWith('error/unhandled-exception');
+				});
+			});
+		});
+	});
+	describe('#postSubmittingFor', () => {
+		describe('When handling a submitting page for post request', () => {
+			describe('and there is an error', () => {
+				const mockErrorValue = {
+					errors: { a: 'b' },
```

Comment:
I think this should assert the error id is correct. 

This checking the errors keys length in the controller. 

A:B is less descriptive than using the coupled error id field. 

you are not asserting the correct id has been used. 

E.G:

```
			describe('and there errors with the body', () => {
				beforeEach(() => {
					req.body = {
						errors: { 'examination-add-another-deadline-item': 'an error' },
						errorSummary: [{ text: 'Error summary', href: '#' }]
					};
					getPageData.mockReturnValue({ text: 'mock page data' });
					postAddAnotherDeadlineItem(req, res);
				});
				it('should render the error page', () => {
					expect(res.render).toHaveBeenCalledWith('pages/examination/add-another-deadline-item', {
						errorSummary: [
							{
								href: '#',
								text: 'Error summary'
							}
						],
						errors: {
							'examination-add-another-deadline-item': 'an error'
						},
						text: 'mock page data'
					});
				});
			});
```

Reply:
This asserts the id from the error should match the id filed used in the error check.